src/app/components/global-styles/global-styles.tsx

```diff
@@ -6,21 +6,29 @@ import { fullPageStyles } from './full-page-styles';
 import { popupStyles } from './popup-styles';
 import { popupCenterStyles } from './popup-center-styles';
 import { tippyStyles } from './tippy-styles';
+import { useThemeSwitcher } from '@app/common/theme-provider';
 
-const styles = css`
-  body {
-    &.no-scroll,
-    &.no-scroll .main-content {
-      overflow: hidden;
+export const GlobalStyles = memo(() => {
+  const [theme] = useThemeSwitcher();
+
+  const styles = css`
```

Comment:
afaik it's better to use the props feature here than declaring inside the component https://emotion.sh/docs/styled#changing-based-on-props ?

Reply:
The `useThemeSwitcher` hook isn't available where `GlobalStyles` is used b/c it is being rendered as a descendant in the same component that renders the provider. Could break it up into two components, and have one call the other and just pass in the props after extracting the theme from the hook, although seems a bit overkill?